File: tests/combi/matrix_intro_calloc.cpp
#include <stdlib.h>
#include "../../include/optitrust.h"

int* q;

const int N = 10;

void allocate (){
 q = (int*) calloc(N, sizeof(int));
}


int main () {

  int* p = (int*) calloc(N, sizeof(int));
  return 0;
}

File: tests/combi/matrix_intro_calloc.ml
open Optitrust
open Target


let _ = Run.doc_script_cpp (fun () -> 

  !! Matrix.intro_calloc [cVarDef "p"];

)

"
#include <stdlib.h>
#include \"../../include/optitrust.h\"

int main(){
  const int N = 10;
  int* p = (int*) calloc (N, sizeof(int));
}
"

let _ = Run.script_cpp (fun _ ->

    !! Matrix.intro_calloc [cVarDef "p"];
    !! Matrix.intro_calloc [cVarDef "q"];

)
